### PR TITLE
Error Logging for Failed Login Attempts

### PR DESCRIPTION
**Pull request by omgitsbillryan**
- Added Sentry exception for a failed login attempt
- Changed `warn` Rails logging to `error`

Failed logins _can be_ normal behavior but I'm realizing that _most of the time_, it's an `error`.  If this gets to be too noisy, we can re-calibrate.

### CODE REVIEW

**Comment by omgitsbillryan** [comment]
Related Issue : https://github.com/department-of-veterans-affairs/vets-api/issues/654

File: app/controllers/v0/sessions_controller.rb [hunk 1]
```diff
@@ -88,5 +85,25 @@ def delete_session(token)
...
+      logger.error user_error
+      logger.error session_error
+
+      if ENV['SENTRY_DSN'].present?
+        Raven.extra_context(
+          saml_response: saml_error,
+          session: session_error,
+          user: user_error
+        )
+        Raven.capture_exception('Login with ID.me failed!')
```
Comment (omgitsbillryan) [inline app/controllers/v0/sessions_controller.rb]: I'm mostly concerned with this piece
- is my usae of `extra_context` ok?
- is my usage of `capture_exception` with just a string ok?

Comment (austin-martinez) [inline app/controllers/v0/sessions_controller.rb]: I think you want to use `capture_message` if the arg is a string
https://docs.sentry.io/clients/ruby/usage/#reporting-messages

Comment (omgitsbillryan) [inline app/controllers/v0/sessions_controller.rb]: I followed the example in the `education_benefits_claims_controller.rb`

```ruby
validation_error = education_benefits_claim.errors.full_messages.join(', ')

Raven.tags_context(validation: 'education_benefits_claim')
Raven.capture_exception(validation_error)
```

CC: @lihanli

Comment (lihanli) [inline app/controllers/v0/sessions_controller.rb]: capture_message seems more appropriate i can change my code

File: app/controllers/v0/sessions_controller.rb [hunk 2]
```diff
@@ -88,5 +85,17 @@ def delete_session(token)
       end
     end
     # :nocov:
+
+    def log_errors
+      message = <<-MESSAGE.strip_heredoc
```
Comment (omgitsbillryan) [inline app/controllers/v0/sessions_controller.rb]: @saneshark i followed your example for preserving tabs, whitespace & newlines [from here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/common/exceptions/external/backend_service_exception.rb#L75), did I get it right?  heredoc is a new concept for me.

Comment (saneshark) [inline app/controllers/v0/sessions_controller.rb]: yes, it looks good to me. All these does is make sure that when you're using the multi-line interpolation, that the leading tabs or spaces are only preserved relative to where to the cursor interpolation block begins or ends, vs the entire line beginning or ending.

**Pull request merged**